### PR TITLE
gammastep: init at 2.0.1

### PR DESCRIPTION
**Pull request by primeos**
###### Motivation for this change

Should work but I need to properly test it (on my test system it crashed but that's probably because the hardware doesn't support gamma adjustments and it looks similar to https://gitlab.com/chinstrap/gammastep/-/issues/3).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by primeos** [comment]
I did test this successfully on Sway now. Gammastep is now recommended in the Sway wiki as Redshift replacement (instead of the previous fork (`redshift-wlr`)):
- https://github.com/swaywm/sway/wiki/i3-Migration-Guide#common-x11-apps-used-on-i3-with-wayland-alternatives
- https://github.com/swaywm/sway/wiki#redshift-equivalent

@globin @yegortimoshenko mind if I reuse the `mkRedshift` for this? It seemed to make sense for now as the expression is quite complex and `gammastep` is a fork of `redshift`.

And we can probably replace `redshift-wlr` with `gammastep` in the future but I wouldn't do it right away.

File: pkgs/applications/misc/redshift/default.nix [hunk 1]
```diff
@@ -62,10 +68,15 @@ let
...
+      postInstall = if (pname == "gammastep") then ''
+        substituteInPlace $out/share/applications/gammastep.desktop \
+          --replace 'Exec=gammastep' "Exec=$out/bin/gammastep"
+        substituteInPlace $out/share/applications/gammastep-indicator.desktop \
+          --replace 'Exec=gammastep-indicator' "Exec=$out/bin/gammastep-indicator"
+      '' else ''
         substituteInPlace $out/share/applications/redshift.desktop \
           --replace 'Exec=redshift' "Exec=$out/bin/redshift"
-        substituteInPlace $out/share/applications/redshift.desktop \
+        substituteInPlace $out/share/applications/redshift-gtk.desktop \
```
Comment (doronbehar) [inline pkgs/applications/misc/redshift/default.nix:79]: This change should come in a dedicated commit such as:

```
redshift: fix substitute command for desktop entry
``

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/redshift/default.nix (hunk 1 above)
Comment (primeos) [reply pkgs/applications/misc/redshift/default.nix:79]: Oh well, I could consider doing that but I don't really see the point tbh. It's a trivial "fix" and it was working perfectly fine without it. Generally I agree, but I'm not sure if it's really worth a separate commit in this case.

**Comment by doronbehar** [comment]
@GrahamcOfBorg build redshift

File: pkgs/applications/misc/redshift/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/applications/misc/redshift/default.nix:79]: Then please add such a comment to the commit message, below `gammastep: init`. And please describe that you reuse `mkRedshift` for `gammastep`.

Comment (primeos) [reply pkgs/applications/misc/redshift/default.nix:79]: I did split the commits this time.

> And please describe that you reuse mkRedshift for gammastep.

IMO commit messages should always explain why, not how (that's what the patch is for).

Also: I appreciate the feedback, I'm just a bit concerned that we waste important time due to [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding) here as all of these changes don't affect the result. But at least it's better documented in the history now :)

**Pull request merged**